packages/desktop/pages/settings.tsx

```diff
@@ -9,6 +9,32 @@ const Page = () => {
   const { isLoading, appConfig, updateAppConfig } = useAppConfig();
   const clientContext = useClientContext();
   const [appVersion, setAppVersion] = useState('');
+  const [featureCode, setFeatureCode] = useState('');
+
+  const parseCode = useCallback(
+    (featureCode: string) => {
```

Comment:
featureCode should be available as a state and don't need to be passed in?

Reply:
this is actually an interesting edge case in react i think -- if creation of your useCallback is slow you can avoid the creation of new functions by passing the input as an arg instead of embedding it in the deps array

pattern came up kind of by accident, that being an arg was just an artifact of a refactor i had done. I'm gonna switch to embedding the featureCode in the deps array since it's just a bit cleaner i think and we don't have any perf concern w/ creation new funcs